### PR TITLE
Support microsecond resolution on access/write times via utimes() sys…

### PR DESCRIPTION
**Pull request by alexischr**
…call

Do not merge yet as a backport of https://github.com/dotnet/corefx/pull/31522 may be used instead

### CODE REVIEW

File: src/Common/src/Interop/Unix/System.Native/Interop.UTimes.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    internal static partial class Sys
+    {
+        internal struct TimeValPair
+        {
+            internal long ASec;  // access time (seconds)
+            internal long AUSec; // access time (microseconds)
+            internal long MSec;  // modified time (seconds)
+            internal long MUSec; // modified time (microseconds)
+        }
```
Comment (Anipik) [inline src/Common/src/Interop/Unix/System.Native/Interop.UTimes.cs]:
```C#
internal struct TimeValPair
{
    internal int Sec;
    internal int USec;
}
```
you can just use an array of this

Comment (Anipik) [reply src/Common/src/Interop/Unix/System.Native/Interop.UTimes.cs]: int in c# is same as long in c

File: src/Native/Unix/System.Native/pal_time.h [hunk 2]
```diff
@@ -16,13 +16,28 @@ typedef struct UTimBuf
     int64_t ModTime;
 } UTimBuf;
 
+typedef struct TimeValPair
+{
```
Comment (Anipik) [inline src/Native/Unix/System.Native/pal_time.h]: i dont think you require this, you can directly use struct timeval

File: src/Native/Unix/System.Native/pal_time.c [hunk 3]
```diff
@@ -38,6 +46,19 @@ int32_t SystemNative_UTime(const char* path, UTimBuf* times)
...
+{
+    assert(times != NULL);
+
+    struct timeval temp [2];
+    ConvertTimeValPair(times, temp);
+
+    int32_t result;
+    while (CheckInterrupted(result = utimes(path, temp)));
+    return result;
+}
```
Comment (Anipik) [inline src/Native/Unix/System.Native/pal_time.c]:
```C
int32_t SystemNative_UTimes(const char* path, struct timeval*  times)
{
    assert(times != NULL);

    int32_t result;
    while (CheckInterrupted(result = utimes(path, times)));
    return result;
}
```

File: src/Native/Unix/System.Native/pal_time.c [hunk 4]
```diff
@@ -26,6 +26,14 @@ static void ConvertUTimBuf(const UTimBuf* pal, struct utimbuf* native)
     native->modtime = (time_t)(pal->ModTime);
 }
 
+static void ConvertTimeValPair(const TimeValPair* pal, struct timeval native[2])
```
Comment (Anipik) [inline src/Native/Unix/System.Native/pal_time.c]: You dont require this too

**Comment by alexischr** [comment]
https://github.com/dotnet/corefx/pull/31522 was used instead

**Pull request closed without merging**